pkg/operator/vspherecontroller/checks/check_nodes.go

```diff
@@ -236,33 +237,40 @@ func getVM(ctx context.Context, checkOpts CheckArgs, node *v1.Node) (*mo.Virtual
 	vmClient := checkOpts.vmConnection.Client.Client
 	vmConfig := checkOpts.vmConnection.Config
 
-	// Find datastore
-	finder := find.NewFinder(vmClient, false)
-	dc, err := finder.Datacenter(ctx, vmConfig.Workspace.Datacenter)
+	dataCenterNames, err := utils.GetDatacenters(vmConfig)
 	if err != nil {
-		return nil, fmt.Errorf("failed to access Datacenter %s: %s", vmConfig.Workspace.Datacenter, err)
+		return nil, err
 	}
 
-	// Find VM reference in the datastore, by UUID
-	s := object.NewSearchIndex(dc.Client())
 	vmUUID := strings.ToLower(strings.TrimSpace(strings.TrimPrefix(node.Spec.ProviderID, "vsphere://")))
+	for _, dcName := range dataCenterNames {
+		// Find datastore
+		finder := find.NewFinder(vmClient, false)
+		dc, err := finder.Datacenter(ctx, dcName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to access Datacenter %s: %s", dcName, err)
+		}
 
-	svm, err := s.FindByUuid(ctx, dc, vmUUID, true, nil)
-	if err != nil {
-		return nil, fmt.Errorf("failed to find VM by UUID %s: %s", vmUUID, err)
-	}
-	if svm == nil {
-		return nil, fmt.Errorf("unable to find VM by UUID %s", vmUUID)
-	}
+		// Find VM reference in the datastore, by UUID
+		s := object.NewSearchIndex(dc.Client())
+
+		svm, err := s.FindByUuid(ctx, dc, vmUUID, true, nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to find VM %s by UUID %s: %s", node.Name, vmUUID, err)
+		}
+		if svm == nil {
+			continue
+		}
```

Comment:
When the VM is not found, will `err == nil` and `svm == nil`?  Won't it return some sort of "not found" error here?

Reply:
Yes that is correct. It doesn't seem to return an error when vm is not found. I have tested it.